wiki/Beatmapping/Health_Drain/en.md

```diff
@@ -1,7 +1,99 @@
+- The slider ball touched the slider end or a reverse arrow successfully (better judgement at the end tick leads to more health gain).
+- Completing a spinner and continuous spinning in the spinner duration (better judgement at the end of the spinner leads to more health gain).
+
+Health loss:
+
+- Passive health drain.
+- Missing a hit circle.
+- Tapping a slider head early.
+- Missing a slider tick, or slider reverse arrow.
+- Failed to achieve at least a 50 on a spinner.
+
+Health reserved:
+
+- Break time.
+
+## osu!taiko
+
+In osu!taiko the health bar starts off empty and must be filled up to at least 50% to achieve a pass on a difficulty. This is done through hitting hit objects in sync with the beat. Score bursts will appear after hitting notes, changing score and health values depending on the accuracy of the hit. A perfectly timed hit (300) will recover health more than a badly timed hit (50). A miss will drain health instead. Failing to complete a denden will result in health loss. Drumrolls can be ignored with no health penalty, as they have no influence on the health bar. Passive health drain is disabled in osu!taiko, so only hit objects affect the health bar. For an overview of health recovery and health loss see below:
+
+Health recovery:
+
+- Getting 'Great' or 'Good' on small/large notes.
+
+Health loss:
+
+- Getting 'Miss' or 'Bad' on small/large notes.
+- Not completing Dendens.
+- Not completing Shakers.
+
+No effect:
+
+- Doing drumrolls.
+- Not doing drumrolls.
+- Completing Dendens.
+- Completing Shakers.
+
+_(Note: It is entirely possible for a beatmap to never pass by design due to it only containing drumrolls and/or dendens/shakers.)_
+
+## osu!catch
+
+In osu!catch health will decrease at a steady rate (passive health drain), but can be replenished by catching fruit and drops. Score bursts will appear after catching fruit, changing score and health values depending on the accuracy of the hit. The player also receives larger boosts if he catches the end of a combo resulting in a [geki](#geki). Missing fruit will drain health instead. Bananas can be ignored with no health penalty. For an overview of health recovery and health loss see below:
+
+Health recovery:
+
+- Collecting any fruits, juice drops, or bananas.
+
+Health loss:
+
+- Constant life drain.
+- Not collecting any fruits or juice drops.
+
+Health reserved:
+
+- None.
+
+## osu!mania
+
+In osu!mania passive health drain is disabled, so only hit objects will affect the health bar. Score bursts will appear after hitting notes, changing score and health values depending on the accuracy of the hit. A perfectly timed hit (300) will recover health more than a badly timed hit (50). The player also receives larger boosts if he achieves a [katu](#katu) or a [geki](#geki) at the end of a combo. A miss will drain health instead. Health bar regeneration occurs slowly while holding hold notes. For an overview of health recovery and health loss see below:
```

Comment:
Does a rainbow 300 restore more health than a regular 300? I honestly don't know but it might be worth mentioning this if it's the case.

Reply:
no idea, cant seem to find any info indicating this on the wiki either

Reply:
had confirmation that rainbow 300s do indeed gain more hp back